atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/ProfilingKeyValueService.java

```diff
@@ -367,12 +388,15 @@ public boolean isInitialized() {
 
     @Override
     public void compactInternally(TableReference tableRef, boolean inMaintenanceMode) {
+        long startTime = System.currentTimeMillis();
         maybeLog(() -> delegate.compactInternally(tableRef, inMaintenanceMode), (logger, stopwatch) -> {
             if (inMaintenanceMode) {
                 // Log differently in maintenance mode - if a compactInternally is slow this might be bad in normal
                 // operational hours, but is probably okay in maintenance mode.
-                logger.log("Call to KVS.compactInternally (in maintenance mode) on table {} took {} ms.",
-                        LoggingArgs.tableRef(tableRef), LoggingArgs.durationMillis(stopwatch));
+                logger.log("Call to KVS.compactInternally (in maintenance mode) at time {}, on table {} took {} ms.",
+                        LoggingArgs.startTimeMillis(startTime),
+                        LoggingArgs.tableRef(tableRef),
+                        LoggingArgs.durationMillis(stopwatch));
             } else {
                 logTimeAndTable("compactInternally", tableRef);
```

Comment:
need to pass the startTime into this method from `compactInternally` as `logTimeAndTable` will get a startTime only when the logger is called.

Reply:
I think the BiConsumers used normally are fine (you call the method, it gets the start time and returns you the BiConsumer with a closure over the start time), it is just here where we call `logTimeAndTable` inside of a BiConsumer (to check maintenance mode) where `logTimeAndTable` is called too late.

Interestingly I think this method may be broken because nothing actually invokes the BiConsumer returned by logTimeAndTable so no logging happens.  This PR should also fix this.  Easiest method is probably inlining the contents of `logTimeAndTable` in this one invokation.